### DISCUSSION TITLE
isDirty problem. ref not update

### DISCUSSION THREAD

**Discussion by Kurochen** [react-hook-form/react-hook-form#3398] [Q&A] [answered]
CodeSandBox - [example](https://codesandbox.io/s/react-hook-form-material-m2cmt?file=/src/App.js)

If we choose "Second", the buttons become active.
If we choose the default value "First", the buttons will be disabled. Cool.

If we choose "Second" and then click cancel, the buttons are disabled.
The value will become "First"
Very cool.

But, if we choose "Second" and then click Cancel, then select "Second" again - the buttons remain disabled.
Moreover, "dirtyData" will be empty.

I think this is all because of the "ref"

**Comment by bluebill1049** [comment, marked as answer]
use `formState.isDirty` and provide a `defaultValues` at `useForm`

**Comment by Kurochen** [comment]
Thanks
`defaultValues` work.
`formState.isDirty` not work quite correctly. In my project. Trying to figure it out.  I have a big form.

**Reply by bluebill1049** [reply]
`isDirty` is driven with form values deepEqual `defaultValues`

**Reply by Kurochen** [reply]
Yes. And there can be many reasons. Unnecessary value in `defaultValues.` May be another type.
I use my first solution.
